src/server/error.h

```diff
@@ -31,6 +31,14 @@ using facade::kWrongTypeErr;
 
 #define RETURN_ON_ERR(x) RETURN_ON_ERR_T(std::error_code, x)
 
+#define RETURN_ON_GENERIC_ERR(x)                                   \
```

Comment:
@adiholden 

I had to, there is no really good solution otherwise. Doing something `RETURN_ON_ERR_T(GenericError, x)` won't work. Yes I tried modifying the micro above by converting the statement: 

`std::error_code __ec = (x); ` ----> `T __ex = (x)` but a) this is not properly being parsed and we get compiler errors, we also do an extra copy in that expression which is fine for `error` code because it's `trivially copyable`. Generic error is not.